engines/bastion_katello/app/assets/javascripts/bastion_katello/docker-tags/details/docker-tag-environments.controller.js

```diff
@@ -11,27 +11,36 @@
  *   Provides the functionality for the docker tags details environments list.
  */
 angular.module('Bastion.docker-tags').controller('DockerTagEnvironmentsController',
-    ['$scope', '$location', 'Nutupane', 'DockerTag', 'CurrentOrganization',
-    function ($scope, $location, Nutupane, DockerTag, CurrentOrganization) {
+    ['$scope', '$location', 'Nutupane', 'DockerTag', 'Repository', 'CurrentOrganization',
+    function ($scope, $location, Nutupane, DockerTag, Repository, CurrentOrganization) {
         var params = {
             'organization_id': CurrentOrganization,
             'search': $location.search().search || "",
             'sort_by': 'name',
             'sort_order': 'ASC',
             'paged': false
         };
-        var nutupane = new Nutupane(DockerTag, params, null, {disableAutoLoad: true});
 
         var renderTable = function () {
-            var ids = _.map($scope.tag.related_tags, 'id');
             var newParams = {
                 'organization_id': CurrentOrganization,
                 'search': $location.search().search || "",
                 'sort_by': 'name',
                 'sort_order': 'ASC',
-                'paged': false,
-                'ids[]': ids
+                'paged': false
             };
+            var ids;
+            var nutupane;
+            if ($scope.tag.repositories.length > 1) {
```

Comment:
I don't understand how we might be looking at a repository *or* a docker tag in this controller. Are we using this controller for multiple views? It feels like we need a separate controller and view for this rather than handle special cases. Let me know if I'm missing the point of your solution here.

Reply:
This is for a single view, which is the environments tab for each individual tag.  So the reason why we need to have two scenarios (tag and repo) is that the environments/CVVs page for a docker tag cannot be created using the same strategy for both Pulp 2 and Pulp 3.  For Pulp 2 each repo gets its own copy of a tag so the `related_tags` method works as intended.  For Pulp 3, however, tags are shared between repos so `related_tags` only ends up returning itself.  My solution was to have this page be "tag-centric" for Pulp 2 and "repo-centric" for Pulp 3 in the background, but have the page look identical to the user.

I see what you mean about splitting this into different controllers, do you think it would work out cleanly to use one bastion_katello controller over another depending on if we're using Pulp 3 or not?  I'm not sure myself at the moment, I'd have to see how bastion_katello chooses which controller to use.  Also since you mentioned maintainability, the Pulp 2 (tag-centric) method of displaying this page will be dropped eventually along with Pulp 2.

Reply:
Now I understand! Thanks to your explanation I think we can keep this front-end changes as you've made them because half of it will be tech-debt which we can clean up when pulp2 is gone